src/main/java/com/avispl/symphony/dal/communicator/ppdswave/PhilipsWaveAggregatorCommunicator.java

```diff
@@ -723,22 +751,135 @@ private void processDeviceDetails() throws Exception {
         }
     }
 
+    /**
+     * Process device input source details. Generate inputSource dropdown with a list of available options
+     * and current value (if specified)
+     *
+     * @param properties to collect device data to
+     * @param controls   to add device controls to
+     * @param deviceNode object containing all the data necessary
+     */
+    private void processDeviceAppContentSources(Map<String, String> properties, List<AdvancedControllableProperty> controls, Display deviceNode) {
+        if (logger.isDebugEnabled()) {
+            logger.debug("Processing device content source: " + deviceNode);
+        }
+        ContentSource contentSource = deviceNode.getContentSource();
+        if (contentSource == null) {
+            return;
+        }
+        List<Source> available = contentSource.getAvailableAppContentSources();
+        if (available == null || available.isEmpty()) {
+            return;
+        }
+        ContentSource.SourceType sourceType = contentSource.getCurrent();
+        String currentValue = sourceType.getValue();
```

Comment:
Do we need to check for null for sourceType here to avoid a potential NPE? If you are sure that it won't cause an error or if you intend to bypass it, then we can omit the null check for sourceType

Reply:
It defaults to new SourceType(Constants.Utility.EMPTY, Constants.Utility.EMPTY) in ContentSource.getCurrent(), so it is considered safe :)